### ISSUE TITLE
Region not detected from credentials file

### ISSUE THREAD

**Issue by mfischer-zd** [aws/aws-sdk-ruby#1073]
The AWS CLI can obtain its working region from a credentials file, e.g.: 

```
[profile tokyo]
source_profile = default
region = ap-northeast-1
role_arn = arn:aws:iam::123456789:role/master-of-the-universe
```

The Ruby code should behave similarly if `AWS_PROFILE` is set, without also requiring `AWS_DEFAULT_REGION` or `AWS_REGION` to be set.

**Comment by awood45** [comment]
Thank you for the feature request. We're looking at the best way to support some of these CLI configuration values going forward.

In the meantime, one option is to use the `AWS_REGION` environment variable, if you use the same region across clients.

**Comment by mfischer-zd** [comment]
Actually, it looks like the Ruby SDK doesn't support the CLI config file format at all ($HOME/.aws/config, as opposed to the legacy $HOME/.aws/credentials file).  See #1045

**Comment by awood45** [comment]
This is correct, that's currently CLI-specific.

Given that there is a large ask for this, we are looking for the best way to support some of these features, without introducing unexpected behavior into the SDK.

**Comment by awood45** [comment]
Addressed by #1178

**Comment by awood45** [comment]
#1178 has been merged, this will be supported with the next release, which will be version `2.4.0`.
